nixos/modules/services/misc/rkvm.nix

```diff
@@ -0,0 +1,168 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  cfg = config.services.rkvm;
+  toml = pkgs.formats.toml { };
+in
+{
+  meta.maintainers = with maintainers; [ ckie ];
+
+  options.services.rkvm = {
+    enable = mkOption {
+      default = cfg.server.enable || cfg.client.enable;
+      defaultText = literalExpression "cfg.server.enable || cfg.client.enable";
+      type = types.bool;
+      description = lib.mdDoc ''
+        Whether to enable rkvm, a Virtual KVM switch for Linux machines.
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.rkvm;
+      defaultText = literalExpression "pkgs.rkvm";
+      description = mdDoc ''
+        The rkvm derivation to use.
+      '';
+    };
+
+    server = {
```

Comment:
Settings for server and client are almost the same.  It would be great if we could avoid this redundancy.  [The frp module](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/networking/frp.nix) might be an example.

Reply:
This was an intentional choice, it allows for better docs and a smaller diff if one component gets more unshared options.